File: client/src/app/shared/services/admin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Room } from '../models/Room';
import { environment } from '../../../environments/environment';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class AdminService {
  private api = `${environment.apiBaseUrl}/admin`;

  constructor(private http: HttpClient) {}

  createRoom(room: Room) {
    return this.http.post<Room>(`${this.api}/rooms`, room);
  }

  getAllRooms() {
    return this.http.get<Room[]>(`${this.api}/rooms`);
  }

  updateRoom(id: number, room: Room) {
    return this.http.put<Room>(`${this.api}/rooms/${id}`, room);
  }

  deleteRoom(id: number) {
    return this.http.delete(`${this.api}/rooms/${id}`);
  }
  // New method to get occupancy count at a specific date and time
  getRoomOccupancy(roomId: number, date: string, time: string): Observable<number> {
    const params = new HttpParams()
      .set('date', date)   // format: 'YYYY-MM-DD'
      .set('time', time);  // format: 'HH:mm:ss' or 'HH:mm'

    return this.http.get<number>(`${this.api}/rooms/${roomId}/occupancy`, { params });
  }


}

File: server/src/main/java/com/coworkingspace/server/services/AdminService.java
package com.coworkingspace.server.services;

import com.coworkingspace.server.DTOs.RoomDTO;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.Map;

public interface AdminService {
    RoomDTO createRoom(RoomDTO dto);
    RoomDTO updateRoom(Long roomId, RoomDTO dto);
    void deleteRoom(Long roomId);
    List<RoomDTO> getAllRooms();
    int getOccupancyCountAt(Long roomId, LocalDate date, LocalTime time);

}
